handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -494,6 +494,12 @@ public void flush(ChannelHandlerContext ctx) throws Exception {
         }
         try {
             wrap(ctx, false);
+        } catch (Exception e) {
+            // Just fail the pending writes in case of an SSLException. If there are no pending writes rethrow.
+            if (pendingUnencryptedWrites.removeAndFailAll(e)) {
```

Comment:
Sorry ... I missed the word `not` in my above statement:

`pendingUnencryptedWrites` is empty -> `pendingUnencryptedWrites` is NOT empty 

I'm curious why we don't always propagate the exception. I guess the rational is if the queue is NOT empty then we failed individual writes with the exception and throwing the exception may be redundant?